src/library/Uno.Material/Styles/Controls/DatePicker.xaml

```diff
@@ -0,0 +1,242 @@
+								</VisualState>
+							</VisualStateGroup>
+
+							<VisualStateGroup x:Name="HasDateStates">
+								<VisualState x:Name="HasDate" />
+								<VisualState x:Name="HasNoDate">
+									<VisualState.Setters>
+										<Setter Target="DayTextBlock.Text"
+												Value="dd" />
+										<Setter Target="DayTextBlock.Foreground"
+												Value="{StaticResource MaterialOnSurfaceLowBrush}" />
+
+										<Setter Target="FirstPickerSpacing.Foreground"
+												Value="{StaticResource MaterialOnSurfaceLowBrush}" />
+
+										<Setter Target="MonthTextBlock.Text"
+												Value="mm" />
+										<Setter Target="MonthTextBlock.Foreground"
+												Value="{StaticResource MaterialOnSurfaceLowBrush}" />
+
+										<Setter Target="SecondPickerSpacing.Foreground"
+												Value="{StaticResource MaterialOnSurfaceLowBrush}" />
+
+										<Setter Target="YearTextBlock.Text"
+												Value="yyyy" />
+										<Setter Target="YearTextBlock.Foreground"
+												Value="{StaticResource MaterialOnSurfaceLowBrush}" />
+									</VisualState.Setters>
+								</VisualState>
+							</VisualStateGroup>
+						</VisualStateManager.VisualStateGroups>
+
+						<!-- Flyout Button -->
+						<Button x:Name="FlyoutButton"
+								Style="{StaticResource DatePickerFlyoutButtonStyle}"
+								IsEnabled="{TemplateBinding IsEnabled}"
+								UseSystemFocusVisuals="{TemplateBinding UseSystemFocusVisuals}"
+								HorizontalAlignment="Stretch"
+								HorizontalContentAlignment="Stretch">
+
+							<Grid Background="{TemplateBinding Background}"
+								  CornerRadius="{TemplateBinding CornerRadius}"
+								  Height="{TemplateBinding Height}">
+
+								<!-- Border -->
+								<Rectangle x:Name="BottomBorder"
+										   Fill="{TemplateBinding BorderBrush}"
+										   VerticalAlignment="Bottom"
+										   Height="2" />
+
+								<!-- Header -->
+								<TextBlock x:Name="HeaderTextBlock"
+										   Text="{TemplateBinding Header}"
+										   Style="{StaticResource MaterialCaption}"
+										   Foreground="{TemplateBinding Foreground}"
+										   HorizontalAlignment="Stretch"
+										   VerticalAlignment="Top"
+										   TextWrapping="Wrap"
+										   Margin="10,8,10,0"
+										   Grid.Row="0" />
+
+								<Grid x:Name="FlyoutButtonContentGrid"
+									  VerticalAlignment="Top"
+									  Margin="10,24,10,0"
+									  Height="24"
+									  Grid.Row="1">
+
+									<Grid.ColumnDefinitions>
+										<ColumnDefinition Width="Auto"
+														  x:Name="DayColumn" />
+										<ColumnDefinition Width="Auto"
+														  x:Name="FirstSpacerColumn" />
+										<ColumnDefinition Width="Auto"
+														  x:Name="MonthColumn" />
+										<ColumnDefinition Width="Auto"
+														  x:Name="SecondSpacerColumn" />
+										<ColumnDefinition Width="Auto"
+														  x:Name="YearColumn" />
+									</Grid.ColumnDefinitions>
+
+									<!-- Day -->
```

Comment:
Wouldn't it be possible to do it all in one textblock?

Reply:
In theory, but I can't mess up with the x:Name